cairo/ethereum/cancun/vm/instructions/environment.cairo

```diff
@@ -395,13 +408,15 @@ func returndatacopy{
         Uint(GasConstants.GAS_VERY_LOW + return_data_copy_gas_cost + extend_memory.value.cost.value)
     );
     if (cast(err, felt) != 0) {
+        EvmImpl.set_stack(stack);
         return err;
     }
 
     let memory = evm.value.memory;
```

Comment:
move this to where it used.
This reference should be next to its usage

Reply:
i dont understand